app/uk/gov/hmrc/taxhistory/model/nps/HIPIabd.scala

```diff
@@ -149,6 +149,11 @@ object HIPIabd {
       case Array(desc, code) => (Some(desc.trim), code.substring(0, code.indexOf(")")).toIntOption)
       case _                 => (None, None)
     }
+    def dateFormating(date: String) = {
```

Comment:
Add error handling for invalid date strings (currently, if parsing fails, it will throw an exception).